packages/salesforcedx-utils-vscode/src/helpers/activationTracker.ts

```diff
@@ -50,24 +50,36 @@ export class ActivationTracker {
   async markActivationStop(activationEndDate?: Date): Promise<void> {
     // capture date and elapsed HR time
     const activateEndDate = activationEndDate ?? new Date();
-    const hrEnd = this.telemetryService.getEndHRTime(this._activationInfo.startActivateHrTime);
+    const hrEnd = this.telemetryService.getEndHRTime(
+      this._activationInfo.startActivateHrTime
+    );
     // getting extension info. This may take up to 10 seconds, as log record creation might be lagging from
     // this code. All needed data have been captured for telemetry, so a wait here should have no effect
     // on quality of telemetry data
     const extensionInfo = await getExtensionInfo(this.extensionContext);
-    // subtract activateEndDate from loadStartDate to get the time spent loading the extension if loadStartDate is not undefined
-    const extensionActivationTime = extensionInfo?.loadStartDate
-      ? activateEndDate.getTime() - extensionInfo.loadStartDate.getTime()
-      : -1;
-
-    this._activationInfo = {
-      ...this._activationInfo,
-      ...extensionInfo,
-      extensionActivationTime,
-      activateEndDate,
-      markEndTime: hrEnd
-    };
-    this.telemetryService.sendActivationEventInfo(this.activationInfo);
+    let extensionActivationTime = -1;
+    if (
+      extensionInfo?.loadStartDate &&
+      activateEndDate.getTime() <= extensionInfo.loadStartDate.getTime()
```

Comment:
Why activateEndDate should be no more than loadStartDate? In my test when loadStartDate is defined, activateEndDate is always larger than loadStartDate, so the extensionActivationTime is a positive value, which makes sense.

Reply:
I have it backward, good catch